framework/include/meshgenerators/MeshGenerator.h

```diff
@@ -203,6 +210,21 @@ MeshGenerator::declareMeshProperty(const std::string & data_name, const T & init
   return data;
 }
 
+template <typename T>
+T &
+MeshGenerator::setMeshProperty(const std::string & data_name, const T & data_value)
+{
```

Comment:
based on what Cody mentioned, you were right adding a limitiation on when this should be run.

Let's allow it in the generate routine only (as you had iirc) since it's likely to error if objects are constructed in the wrong order and this relies on a previous object having declared the routine

Reply:
Added the error condition back